### PR TITLE
Fixes on general usages of stat

### PR DESCRIPTION
**Pull request by otaviobp**
Trying to prevent here to run stat + file operation. Prefer to run the file operation and on errors evaluate the result.

### CODE REVIEW

File: src/lib/sys.c [hunk 1]
```diff
@@ -244,6 +255,13 @@ bool file_is_executable(const char *filename)
 	return false;
 }
 
+bool sys_file_is_hardlink(const char *file1, const char *file2)
+{
+	struct stat sb, sb2;
+
+	return !(lstat(file1, &sb) == 0 && lstat(file2, &sb2) == 0 && sb.st_ino == sb2.st_ino);
```
Comment (castulo) [inline src/lib/sys.c]: why are you negating this? if both inodes are the same doesn't that mean it is a hardlink?

File: src/lib/sys.c [hunk 2]
```diff
@@ -348,3 +366,86 @@ bool is_dir(const char *path)
...
+static int sys_rmdir(const char *path)
+{
+	if (rmdir(path) != 0) {
+		return -errno;
+	}
+
+	return 0;
+}
+
+static int sys_rm_dir_recursive(const char *path)
```
Comment (castulo) [inline src/lib/sys.c]: I see the recursiveness will take care of other directories within the directory, but what if any of the directories still have files in it?

File: src/globals.c [hunk 3]
```diff
@@ -311,20 +311,23 @@ bool set_path_prefix(char *path)
...
 	}
-	if (ret < 0 || !S_ISDIR(statbuf.st_mode)) {
+	if (!new_path) {
 		goto error;
 	}
 
-	new_path = realpath(path, NULL);
-	if (!new_path) {
+	ret = stat(new_path, &statbuf);
+	// If path doesn't exit tries to create it
```
Comment (castulo) [inline src/globals.c]: this comment is now in the wrong section of the code

File: src/lib/sys.c (hunk 1 above)
Comment (otaviobp) [reply src/lib/sys.c]: Wow, this is wrong. Not is wrong here and where I call it, so it's working, but I'll fix that.

File: src/lib/sys.c (hunk 2 above)
Comment (otaviobp) [reply src/lib/sys.c]: we call sys_rm_recursive() that will remove files too.

File: src/globals.c (hunk 3 above)
Comment (otaviobp) [reply src/globals.c]: True. I'll move it

**Comment by otaviobp** [comment]
Fixed the reported problems and rebased on master

**Pull request merged**